tools/scripts/add_library_tso_doublets.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+import anndata as ad
+import pandas as pd
+from sklearn.neighbors import KNeighborsTransformer
+import argparse
+import scanpy as sc
+
+
+# Function to flag cell barcodes that are filtered as cells by STARsolo
+def call_cells(cellbarcodes, gex_h5ad):
+    cells=pd.read_csv(cellbarcodes, sep="\t", header=None)
+    adata=ad.read_h5ad(gex_h5ad)
+    adata.obs["star_IsCell"] = False
+    adata.obs.loc[adata.obs.index.isin(cells[0]), 'star_IsCell'] = True
+    return adata
+
+# Function to compute doublet scores using a modified version of DoubletFinder
+# This python implementation was provided by the Allen Institute
+def compute_doublet_scores(gex_h5ad_modified, proportion_artificial=0.2):
+    adata = gex_h5ad_modified
+    adata.var_names_make_unique()
+    adata = adata[adata.obs["star_IsCell"] == True, :]
+    print("adata with star_IsCell == True", adata)
+    k = np.int64(np.round(np.min([100, adata.shape[0] * 0.01])))
```

Comment:
Use `int()` instead of `np.int64()` for better compatibility

Reply:
revised to include